src/tmpfiles/tmpfiles.c

```diff
@@ -2632,6 +2729,58 @@ static int find_gid(const char *group, gid_t *ret_gid, Hashmap **cache) {
         return name_to_gid_offline(arg_root, group, ret_gid, cache);
 }
 
+static int parse_age_by_from_arg(const char *age_by_str, Item *item) {
+        AgeBy ab_f = 0, ab_d = 0;
+
+        static const struct {
+                char age_by_chr;
+                AgeBy age_by_flag;
+        } age_by_types[] = {
+                { 'a', AGE_BY_ATIME },
+                { 'b', AGE_BY_BTIME },
+                { 'c', AGE_BY_CTIME },
+                { 'm', AGE_BY_MTIME },
+        };
+
+        assert(age_by_str);
+        assert(item);
+
+        if (isempty(age_by_str))
+                return -EINVAL;
+
+        for (const char *s = age_by_str; *s != 0; s++) {
+                /* Ignore whitespace. */
+                if (strchr(WHITESPACE, *s))
+                        continue;
+
+                size_t i;
+                for (i = 0; i < ELEMENTSOF(age_by_types); i++) {
+                        /* Check lower-case for files, upper-case for directories. */
+                        if (*s == age_by_types[i].age_by_chr) {
+                                SET_FLAG(ab_f, age_by_types[i].age_by_flag, true);
```

Comment:
SET_FLAG() is great if the third argument is a variable, i.e. sometimes turns the flag off, and on. But if you just specify a literal constant, then there's really not much pont in the macro. just use:

```c
ab_f |= age_by_types[i].age_by_flag;
```

I think this is easier to read without additional info for the typical C programmer...

Reply:
ACK.